src/core/server/legacy/config/__snapshots__/legacy_object_to_config_adapter.test.ts.snap

```diff
@@ -119,7 +119,10 @@ Object {
 
 exports[`#set correctly sets values for paths that do not exist. 1`] = `
 Object {
-  "unknown": "value",
+  "unknown": Object {
+    "sub1": "sub-value-1",
+    "sub2": "sub-value-2",
+  },
```

Comment:
It appears this changed when switching from lodash3's `set` to lodash4's in `src/core/server/config/object_to_config_adapter.ts`.

From what I can tell looking at the test, this seems to be more correct behavior than with lodash3?

```ts
  test('correctly sets values for paths that do not exist.', () => {
    const configAdapter = new LegacyObjectToConfigAdapter({});

    configAdapter.set('unknown', 'value');
    configAdapter.set(['unknown', 'sub1'], 'sub-value-1');
    configAdapter.set('unknown.sub2', 'sub-value-2');

    expect(configAdapter.toRaw()).toMatchSnapshot();
  });
```

@elastic/kibana-platform do we anticipate any breakages from this? I suspect not since this is only used by legacy plugins (which we have few of at this point).

Reply:
@joshdover I realise that during the upgrade but my understand was the same as yours. The new behaviour looks more correct than the old one that's why I've updated the snapshot. Thanks for catching this and bring up the discussion around it!

Reply:
Shouldn't be a problem